src/daft-local-execution/src/intermediate_ops/hash_join_probe.rs

```diff
@@ -14,58 +16,149 @@ use crate::pipeline::PipelineResultType;
 
 enum HashJoinProbeState {
     Building,
-    ReadyToProbe(Arc<ProbeTable>, Arc<Vec<Table>>),
+    ReadyToProbe(Arc<dyn Probeable>, Arc<Vec<Table>>),
 }
 
 impl HashJoinProbeState {
-    fn set_table(&mut self, table: &Arc<ProbeTable>, tables: &Arc<Vec<Table>>) {
+    fn set_table(&mut self, table: &Arc<dyn Probeable>, tables: &Arc<Vec<Table>>) {
         if let HashJoinProbeState::Building = self {
             *self = HashJoinProbeState::ReadyToProbe(table.clone(), tables.clone());
         } else {
             panic!("HashJoinProbeState should only be in Building state when setting table")
         }
     }
 
-    fn probe(
+    fn probe_inner(
         &self,
         input: &Arc<MicroPartition>,
-        right_on: &[ExprRef],
-        pruned_right_side_columns: &[String],
+        probe_on: &[ExprRef],
+        common_join_keys: &[String],
+        left_side_non_join_columns: &[String],
+        right_side_non_join_columns: &[String],
+        build_on_left: bool,
     ) -> DaftResult<Arc<MicroPartition>> {
         if let HashJoinProbeState::ReadyToProbe(probe_table, tables) = self {
             let _growables = info_span!("HashJoinOperator::build_growables").entered();
 
-            // Left should only be created once per probe table
-            let mut left_growable =
+            let mut build_side_growable =
```

Comment:
Haven't actually tested the performance, but my theory is that the growable will have a smaller memory foot print than doing a take. Reason being, if we do the take method, we'd actually have to do a concat first, and this could be quite costly, whereas for the growable, we can elide this concat and leave the Vec<Tables> as is. 

We should definitely test this, but, since we already see pretty good performance with the growable method it's probably not a huge priority

Reply:
And yes you're right we should set the capacity for left/rights